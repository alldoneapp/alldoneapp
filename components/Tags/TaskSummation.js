import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import styles, { colors, windowTagStyle } from '../styles/global'
import Icon from '../Icon'
import { useSelector } from 'react-redux'
import { translate } from '../../i18n/TranslationService'
import { getEstimationResume } from '../../utils/EstimationHelper'

const TaskSummation = ({ projectId, estimation, style, onPress, isMobile, outline, disabled }) => {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)
    const estimationResume = getEstimationResume(projectId, estimation)

    const getText = () => {
        if (outline || smallScreenNavigation || isMobile) {
            if (estimationResume.text.startsWith('P')) {
                return ''
            } else {
                return ` ${translate(`Initial of ${estimationResume.text}`)}`
            }
        } else {
            const hours = estimationResume.hours > 0 ? ` (${estimationResume.hours} ${translate('hours')})` : ''
            return ` ${translate(estimationResume.text)}${hours}`
        }
    }

    return estimation > 0 ? (
        <TouchableOpacity onPress={onPress} disabled={disabled}>
            <View style={[(outline ? otl : localStyles).container, style]}>
                <Icon
                    name={'summation'}
                    size={outline ? 14 : 16}
                    color={outline ? colors.UtilityBlue200 : colors.Text03}
                    style={(outline ? otl : localStyles).icon}
                />
                <Text style={[(outline ? otl : localStyles).text, windowTagStyle()]}>{`${
                    estimationResume.value
                }${getText()}`}</Text>
            </View>
        </TouchableOpacity>
    ) : null
}

const localStyles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: colors.Gray300,
        borderRadius: 12,
        alignItems: 'center',
        justifyContent: 'center',
        height: 24,
    },
    icon: {
        marginHorizontal: 4,
    },
    text: {
        ...styles.subtitle2,
        color: colors.Text03,
        marginVertical: 1,
        marginRight: 10,
        marginLeft: 2,
    },
})

const otl = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: 'transparent',
        borderRadius: 50,
        borderWidth: 1,
        borderColor: colors.UtilityBlue200,
        alignItems: 'center',
        justifyContent: 'center',
        height: 20,
    },
    icon: {
        marginHorizontal: 3,
    },
    text: {
        ...styles.caption1,
        color: colors.UtilityBlue200,
        marginVertical: 1,
        marginRight: 6,
        marginLeft: 2,
    },
})

export default TaskSummation
