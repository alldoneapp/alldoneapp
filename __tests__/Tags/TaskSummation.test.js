/**
 * @jest-environment jsdom
 */

import React from 'react'
import TaskSummation from '../../components/Tags/TaskSummation'

import renderer from 'react-test-renderer'

describe('Task Summation Estimation tag component', () => {
    describe('Task Summation snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<TaskSummation estimation={0} style={{ marginLeft: 10 }} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
        it('should render correctly for amount 1', () => {
            const tree = renderer.create(<TaskSummation estimation={1} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
        it('should render correctly for amount 2', () => {
            const tree = renderer.create(<TaskSummation estimation={2} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('Function updateState snapshot test', () => {
        it('should execute and render correctly', () => {
            const tree = renderer.create(<TaskSummation estimation={1} />)
            expect(tree.toJSON()).toMatchSnapshot()

            tree.getInstance().updateState()
            expect(tree.toJSON()).toMatchSnapshot()
        })
    })

    describe('Task Sub Tasks tag check unmount', () => {
        it('should unmount correctly', () => {
            const tree = renderer.create(<TaskSummation estimation={2} />)
            tree.getInstance().componentWillUnmount()
        })
    })
})
