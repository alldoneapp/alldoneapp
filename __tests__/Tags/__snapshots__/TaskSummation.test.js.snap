// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Summation Estimation tag component Function updateState snapshot test should execute and render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7ECEF",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 16,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.16,
            "lineHeight": 22,
          },
          Object {
            "color": "#8A94A6",
            "marginLeft": 2,
            "marginRight": 10,
            "marginVertical": 1,
          },
        ]
      }
    >
      1 Point
    </Text>
  </View>
</View>
`;

exports[`Task Summation Estimation tag component Function updateState snapshot test should execute and render correctly 2`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7ECEF",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 16,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.16,
            "lineHeight": 22,
          },
          Object {
            "color": "#8A94A6",
            "marginLeft": 2,
            "marginRight": 10,
            "marginVertical": 1,
          },
        ]
      }
    >
      1 Point
    </Text>
  </View>
</View>
`;

exports[`Task Summation Estimation tag component Task Summation snapshot test should render correctly 1`] = `null`;

exports[`Task Summation Estimation tag component Task Summation snapshot test should render correctly for amount 1 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7ECEF",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 16,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.16,
            "lineHeight": 22,
          },
          Object {
            "color": "#8A94A6",
            "marginLeft": 2,
            "marginRight": 10,
            "marginVertical": 1,
          },
        ]
      }
    >
      1 Point
    </Text>
  </View>
</View>
`;

exports[`Task Summation Estimation tag component Task Summation snapshot test should render correctly for amount 2 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7ECEF",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 16,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.16,
            "lineHeight": 22,
          },
          Object {
            "color": "#8A94A6",
            "marginLeft": 2,
            "marginRight": 10,
            "marginVertical": 1,
          },
        ]
      }
    >
      2 Points
    </Text>
  </View>
</View>
`;
